allen-cell-animated/simularium-viewer
# Feature/local custom geo

Problem
=======
resolves #177 

Solution
========
If additional files are dropped along with the simularium file, cache the contents, and then look in the cache for data before making requests for data hosted remotely. 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)
* This change requires updated or new tests

Change summary:
---------------
* Pass additional files in as a mapping of file names to text strings
* add additional registries to store cached data 
* check cache before loading from url 

Steps to Verify:
----------------
1. npm start
2. unzip the file I'll attach 
3. drop the contents of the folder (grab the individual files) and drop them all together
4. Should load most of the geo except for the one that says "should fail"

Screenshots (optional):
-----------------------
Some of the geometry was loaded via url, some was dropped onto the viewer, and one is missing on purpose, so shows spheres. 
<img width="684" alt="Screen Shot 2021-08-25 at 3 05 55 PM" src="https://user-images.githubusercontent.com/5170636/130881884-48d0c7ca-c3e1-4d46-81b3-91d569ee2b9f.png">


Thanks for contributing!
